async_upnp_client/server.py

```diff
@@ -103,6 +109,118 @@ class NopRequester(UpnpRequester):  # pylint: disable=too-few-public-methods
     """NopRequester, does nothing."""
 
 
+class EventSubscriber:
+    """Represent a service subscriber."""
+
+    DEFAULT_TIMEOUT = 3600
+
+    def __init__(self, callback_url: str, timeout: Optional[int]) -> None:
+        """Initialize."""
+        self._url = callback_url
+        self._uuid = str(uuid4())
+        self._event_key = 0
+        self._expires = datetime.now()
+        self.timeout = timeout
+
+    @property
+    def url(self) -> str:
+        """Return callback URL."""
+        return self._url
+
+    @property
+    def uuid(self) -> str:
+        """Return subscriber uuid."""
+        return self._uuid
+
+    @property
+    def timeout(self) -> Optional[int]:
+        """Return timeout in seconds."""
+        return self._timeout
+
+    @timeout.setter
+    def timeout(self, timeout: Optional[int]) -> None:
+        """Set timeout before unsubscribe."""
+        if timeout is None:
+            timeout = self.DEFAULT_TIMEOUT
+        self._timeout = timeout
+        self._expires = datetime.now() + timedelta(seconds=timeout)
+
+    @property
+    def expiration(self) -> datetime:
+        """Return expiration time of subscription."""
+        return self._expires
+
+    def get_next_seq(self) -> int:
+        """Return the next sequence number for an event."""
+        res = self._event_key
+        self._event_key += 1
+        if self._event_key > 0xFFFF_FFFF:
+            self._event_key = 1
+        return res
+
+
+class UpnpEventableStateVariable(UpnpStateVariable):
+    """Representation of an eventable State Variable."""
+
+    def __init__(
+        self, state_variable_info: StateVariableInfo, schema: vol.Schema
+    ) -> None:
+        """Initialize."""
+        super().__init__(state_variable_info, schema)
+        self._last_sent = datetime.fromtimestamp(0, timezone.utc)
+        self._defered_event: Optional[asyncio.TimerHandle] = None
+        self._sent_event = asyncio.Event()
+
+    @property
+    def event_triggered(self) -> asyncio.Event:
+        """Return event object for trigger completion."""
+        return self._sent_event
+
+    @property
+    def max_rate(self) -> float:
+        """Return max event rate."""
+        type_info = cast(
+            EventableStateVariableTypeInfo, self._state_variable_info.type_info
+        )
+        return type_info.max_rate or 0.0
+
+    @property
+    def value(self) -> Optional[T]:
+        """Get the value, python typed."""
+        if self._value is UpnpStateVariable.UPNP_VALUE_ERROR:
+            return None
+
+        return self._value
+
+    @value.setter
+    def value(self, value: Any) -> None:
```

Comment:
I think you can/should call the super `value` setter. The same for the `value` getter.